src/Provider/PhoneNumberProvider.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\sms\Provider\PhoneNumberProvider.
+ */
+
+namespace Drupal\sms\Provider;
+
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Utility\Token;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\sms\Exception\PhoneNumberSettingsException;
+use Drupal\sms\Message\SmsMessageInterface;
+use Drupal\sms\Message\SmsMessage;
+use Drupal\Component\Utility\Random;
+use Drupal\sms\Exception\NoPhoneNumberException;
+
+/**
+ * Phone number provider.
+ */
+class PhoneNumberProvider implements PhoneNumberProviderInterface {
+
```

Comment:
The name doesn't well capture its functionality. A more explanatory description on the class doc would be useful.